src/domain/publicPortal/create.js

```diff
@@ -0,0 +1,25 @@
+const { Portal } = require('../../infra/database/models');
+const { findOne } = require('./find');
+
+async function create(fileId, contractAddress, files, resolvedContent, resolvedMetadata) {
+    const portal = await new Portal({ fileId, contractAddress, files, resolvedContent, resolvedMetadata }).save();
+    return portal;
+}
+
+async function update(fileId, contractAddress, files) {
+    const portal = await Portal.findOneAndUpdate({ fileId, contractAddress }, { files });
+    return portal;
+}
+
+async function getPortalOps(fileId, contractAddress) {
+    const portal = await findOne(fileId, contractAddress);
+    return portal ? update : create;
```

Comment:
I don't think you need two separate functions for this `findOneAndUpdate` has an option to create new if it does not find data in db
https://mongoosejs.com/docs/tutorials/findoneandupdate.html

Reply:
oh wow. Didn't know that. 
Removing the other func.